### DISCUSSION TITLE
Why does property destructuring work here?

### DISCUSSION THREAD

**Discussion by kubk** [mobxjs/mobx#2630] [Q&A] [answered]
I am used to think that JS destructuring shouldn't be used with Mobx, because Mobx tracks property access, not values. For example this code will not react:
```js
let { title } = message
autorun(() => {
    console.log(title)
})
message.updateMessage("Bar")
```
I am wondering why does this example work: https://codesandbox.io/s/busy-https-o3jyd?file=/src/App.js
When I switch from a class store to a functional one it stops to work: https://codesandbox.io/s/dazzling-bash-oht7o?file=/src/App.js

**Comment by mweststrate** [comment]
See https://mobx.js.org/understanding-reactivity.html, you're example is
quite literally in there

**Comment by urugator** [comment, marked as answer]
Try to `console.log(this)` inside the `increment` and you will see whats wrong
(the other one works, because it's using arrow function)
